src/main/java/seedu/address/model/person/Date.java

```diff
@@ -47,12 +50,32 @@ public Date(LocalDate date) {
     }
 
     /**
-     * Returns true if a given string is a valid date.
+     * Returns true if a given string is a valid date format.
      */
-    public static boolean isValidDate(String test) {
+    public static boolean isValidDateFormat(String test) {
         return test.matches(VALIDATION_REGEX);
     }
 
+    /**
+     * Returns true if a given string is a valid date.
+     */
+    public static boolean isValidDate(String fullDate) {
```

Comment:
To check for a valid date would something like this be a better solution?

https://stackoverflow.com/questions/61966525/check-if-date-is-valid-java

Reply:
I thought about it but I felt like it is not really in the spirit of the project since Message constraint is used other than in this class